src/index.js

```diff
@@ -368,6 +370,110 @@ export class NetworkInfo {
 		this::_ensureDefaultOptions();
 	}
 
+	async applySettings (name, settings) {
+		if (_isNullOrUndefined(name)) {
+			throw new Error('interface name is required');
+		}
+
+		if (_isNullOrUndefined(settings)) {
+			throw new Error('settings are required');
+		}
+
+		// get the interface... if it does not exist, this will result in error
+		let
+			iface = await this::_ifconfig(name),
+			result;
+
+		// check to see if adapter should be enabled / disabled
+		if (!_isNullOrUndefined(settings.active)) {
+			debug(
+				'attempt %sabling interface %s',
+				settings.active ? 'en' : 'dis',
+				name);
+
+			if (settings.active) {
+				result = await this::_ifconfig(name, DEFAULT_ACTIVE_UP);
+			} else {
+				result = await this::_ifconfig(name, DEFAULT_ACTIVE_DOWN);
+			}
+
+			debug(
+				'result of %sabling %s: %o',
+				settings.active ? 'en' : 'dis',
+				name,
+				result);
+		}
+
+		// check for hardware address update
+		if (!_isNullOrUndefined(settings.hardwareAddress)) {
+			debug('attempt setting hardware address for interface %s', name);
+
+			result = await this::_ifconfig(
+				name,
+				'hw',
+				'ether',
+				settings.hardwareAddress);
+
+			debug(
+			'result of setting hardware address for interface %s: %o',
+			name,
+			result);
+		}
+
+		// check for ipv4 settings updates
+		if (!_isNullOrUndefined(settings.ipv4)) {
+			debug('attempt ipv4 settings update');
+
+			let
+				args = [name],
+				addr = settings.ipv4;
+
+			// if an array, select the first object
+			if (Array.isArray(addr)) {
+				addr = addr[0];
+			}
+
+			if (!_isNullOrUndefined(addr.address)) {
+				debug('applying ipv4 hardware address: %s', addr.address);
+				args.push(addr.address);
+			}
+
+			if (!_isNullOrUndefined(addr.broadcast)) {
+				debug('applying ipv4 hardware broadcast: %s', addr.broadcast);
+				args.push('broadcast', addr.broadcast);
+			}
+
+			if (!_isNullOrUndefined(addr.netmask)) {
+				debug('applying ipv4 hardware netmask: %s', addr.netmask);
+				args.push('netmask', addr.netmask);
+			}
+
+			result = await this::_ifconfig(...args);
```

Comment:
is it possible to get here with no ip, broadcast or netmask?

Reply:
Yes - I need to add a check against the length of the `args` array... good catch